assets/source/setup-guide/app/steps/SetupTracking.js

```diff
@@ -283,10 +286,16 @@ const SetupTracking = ( { view = 'settings' } ) => {
 	};
 
 	const handleCompleteSetup = async () => {
+		await connectAdvertiser(
+			appSettings.tracking_advertiser,
+			appSettings.tracking_tag
+		);
+
 		recordEvent( 'pfw_setup', {
 			target: 'complete',
 			trigger: 'setup-tracking-complete',
 		} );
+
 		// Force reload WC admin page to initiate the relevant dependencies of the Dashboard page.
 		const path = getNewPath( {}, '/pinterest/settings', {} );
```

Comment:
🚧 In `connectAdvertiser`, we have a try-catch block around the API call. If there is an error in the API call, we will catch the error, display an error notice, _and then_ return to this `handleCompleteSetup` function and call `recordEvent` and proceed to the new page using `window.location`, which is not desired.

Another thing to note is that `connectAdvertiser` is called from two places: `SetupTracking.js` without try-catch block, and `SaveSettingsButton.js` with try-catch block. I think it would be good to make both places similar.

I can think of two ideas to solve this: 

Idea 1: In `useConnectAdvertiser`, we catch the error and re-throw the error again, so that the parent component is able to know there is an error in calling `await connectAdvertiser`.

Idea 2: We don't need `useConnectAdvertiser` hook that wraps around `useCreateNotice`. Instead, we just need a regular `connectAdvertiser` function that wraps around the `await apiFetch` API call, and let the parent component call the function in the try-catch block and display the error notice. `SetupTracking.js` and `SaveSettingsButton.js` already have `createNotice`.

Personally, I'm leaning towards idea 2. Feel free to implement other ideas you see fit. 🙂